assembl/graphql/discussion.py

```diff
@@ -17,18 +17,198 @@
+
+    def resolve_button_label_entries(self, args, context, info):
+        discussion_id = context.matchdict['discussion_id']
+        discussion = models.Discussion.get(discussion_id)
+        if discussion.subtitle:
+            return resolve_langstring_entries(discussion, 'button_label')
+
+        return []
+
+    def resolve_header_image(self, args, context, info):
+        LANDING_PAGE_HEADER_IMAGE = models.AttachmentPurpose.LANDING_PAGE_HEADER_IMAGE.value
+        discussion_id = context.matchdict['discussion_id']
+        discussion = models.Discussion.get(discussion_id)
+        for attachment in discussion.attachments:
+            if attachment.attachmentPurpose == LANDING_PAGE_HEADER_IMAGE:
+                return attachment.document
+
+    def resolve_logo_image(self, args, context, info):
+        LANDING_PAGE_LOGO_IMAGE = models.AttachmentPurpose.LANDING_PAGE_LOGO_IMAGE.value
+        discussion_id = context.matchdict['discussion_id']
+        discussion = models.Discussion.get(discussion_id)
+        for attachment in discussion.attachments:
+            if attachment.attachmentPurpose == LANDING_PAGE_LOGO_IMAGE:
+                return attachment.document
+
+
+class UpdateDiscussion(graphene.Mutation):
+    class Input:
+        title_entries = graphene.List(LangStringEntryInput)
+        subtitle_entries = graphene.List(LangStringEntryInput)
+        button_label_entries = graphene.List(LangStringEntryInput)
+        header_image = graphene.String()
+        logo_image = graphene.String()
+
+    discussion = graphene.Field(lambda: Discussion)
+
+    @staticmethod
+    @abort_transaction_on_exception
+    def mutate(root, args, context, info):
+        cls = models.Discussion
+        discussion_id = context.matchdict['discussion_id']
+        discussion = cls.get(discussion_id)
+        user_id = context.authenticated_userid or Everyone
+
+        permissions = get_permissions(user_id, discussion_id)
+        allowed = discussion.user_can(
+            user_id, CrudPermissions.UPDATE, permissions)
+        if not allowed:
+            raise HTTPUnauthorized()
+
+        with cls.default_db.no_autoflush as db:
+            title_entries = args.get('title_entries')
+            if title_entries is not None and len(title_entries) == 0:
+                raise Exception(
+                    'Title entries needs at least one entry')
+                # Better to have this message than
+                # 'NoneType' object has no attribute 'owner_object'
+                # when creating the saobj below if title=None
+
+            update_langstring_from_input_entries(
+                discussion, 'title', title_entries)
+
+            subtitle_entries = args.get('subtitle_entries')
+            update_langstring_from_input_entries(
+                discussion, 'subtitle', subtitle_entries)
+
+            button_label_entries = args.get('button_label_entries')
```

Comment:
what happens if `button_label_entries` or `subtitle_entries` are `None`?
Now we try to set required on most fields for mutations when it makes sense (better for generated flow types also)

Reply:
discussion.button_label will stay untouched, so None. It will be also None if button_label_entries equals `[]`.